CoreScriptsRoot/CoreScripts/PurchasePromptScript3.lua

```diff
@@ -1660,12 +1660,12 @@ end
 
 local function onVREnabled(vrEnabled) 
 	if vrEnabled then
-		IsVRMode = true
 		local Dialog = require(RobloxGui.Modules.VR.Dialog)
 		if not purchaseDialogVR then
 			purchaseDialogVR = Dialog.new()
 		end
 		purchaseDialogVR:SetContent(PurchaseDialog)
+		IsVRMode = true
```

Comment:
I moved it down so that if requiring Dialog yields (it WaitForChild's on some stuff), you won't get an error trying to index purchaseDialogVR elsewhere.